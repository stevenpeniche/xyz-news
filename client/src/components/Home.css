.home {
  min-height: 860px;
  display: flex;
  align-items: center;
  justify-content: center;
}