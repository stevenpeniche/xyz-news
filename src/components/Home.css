.home {
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
}

.sources {
  height: 100px;
  width: 350px;
  display: flex;
  align-items: center;
  justify-content: center;
	-moz-border-radius: 10px;
	-webkit-border-radius: 10px;
	border-radius: 10px;
  background: #f5f5f5;
  box-shadow: 0 3px 6px rgba(0,0,0,0.16), 0 3px 6px rgba(0,0,0,0.23);
}

.source {
  display: inline-block;
  text-decoration: none;
  color: #000000;
  font-family: 'Roboto', sans-serif;
}

.source-icon {
  width: 60px;
  height: 60px;
  margin: auto;
	background: #ff6600;
	-moz-border-radius: 50px;
	-webkit-border-radius: 50px;
  border-radius: 50px;
  box-shadow: 0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24);
  transition: all 0.3s cubic-bezier(.25,.8,.25,1);
  text-align: center;
}

.source-icon:hover {
  box-shadow: 0 14px 28px rgba(0,0,0,0.25), 0 10px 10px rgba(0,0,0,0.22);
}

.hacker-news-acronym {
  position: relative;
  top: 20px;
  font-size: 22px;
  color: #ffffff;
}

.hacker-news-title {
  font-size: 14px;
}