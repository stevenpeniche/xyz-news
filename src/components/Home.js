import React from 'react';
import Sources from './Sources';

const Home = (props) => {
  return (
    <div>
      <Sources sources={props.sources}/>
    </div>
  );
}

export default Home;