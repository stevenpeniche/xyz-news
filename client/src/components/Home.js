import React from 'react';
import Sources from './Sources';
import './Home.css';

const Home = (props) => {
  return (
    <div className="home">
      <Sources sources={props.sources}/>
    </div>
  );
}

export default Home;