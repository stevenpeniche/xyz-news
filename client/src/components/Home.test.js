import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Home from './Home';

let sources, wrapper;

beforeAll(() => {
  sources = [
    {
      id: 'bleacher-report',
      title: 'Bleacher Report',
      faviconURL: 'https://besticon-demo.herokuapp.com/icon?url=https://bleacherreport.com/&size=70..120..200',
      primaryColor: '#000000',
      topics: [
        {
          name: 'Top'
        },
        {
          name: 'MLB'
        },
        {
          name: 'NFL'
        },
        {
          name: 'NBA'
        },
        {
          name: 'UFC'
        },
        {
          name: 'Boxing'
        },
        {
          name: 'WWE'
        }
      ]
    },
    {
      id: 'cnn',
      title: 'CNN',
      faviconURL: 'https://besticon-demo.herokuapp.com/icon?url=https://bleacherreport.com/&size=70..120..200',
      primaryColor: '#cc0000',
      topics: [
        {
          name: 'Top'
        },
        {
          name: 'World'
        },
        {
          name: 'Politics'
        },
        {
          name: 'Money'
        },
        {
          name: 'Health'
        },
        {
          name: 'Travel'
        },
        {
          name: 'Tech'
        }
      ]
    },
    {
      id: 'vice-news',
      title: 'Vice News',
      faviconURL: 'https://besticon-demo.herokuapp.com/icon?url=https://bleacherreport.com/&size=70..120..200',
      primaryColor: '#000000',
      topics: [
        {
          name: 'Top'
        },
        {
          name: 'Drugs'
        },
        {
          name: 'Tech'
        },
        {
          name: 'Money'
        },
        {
          name: 'Immigration'
        },
        {
          name: 'Politics'
        },
        {
          name: 'Terrorism'
        }
      ]
    }
  ];

  wrapper = shallow( <Home sources={sources} />);
});

describe('Home rendering', () => {
  it('renders without crashing', () => {
    wrapper
  });

  it('renders correctly', () => {
    const tree =
      renderer.create(
        <MemoryRouter>
          <Home sources={sources} />
        </MemoryRouter>
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});