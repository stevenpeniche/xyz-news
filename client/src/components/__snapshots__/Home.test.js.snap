// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home rendering renders correctly 1`] = `
<div
  className="home"
>
  <div
    className="sources animated fadeInDown"
  >
    <a
      className="source animated fadeIn"
      href="/bleacher-report"
      onClick={[Function]}
    >
      <div
        className="source-icon-container"
      >
        <img
          alt="Bleacher Report"
          className="source-icon levitate"
          src="https://besticon-demo.herokuapp.com/icon?url=https://bleacherreport.com/&size=70..120..200"
          style={
            Object {
              "backgroundColor": "#000000",
            }
          }
        />
      </div>
      <div
        className="source-title"
      >
        Bleacher Report
      </div>
    </a>
    <a
      className="source animated fadeIn"
      href="/cnn"
      onClick={[Function]}
    >
      <div
        className="source-icon-container"
      >
        <img
          alt="CNN"
          className="source-icon levitate"
          src="https://besticon-demo.herokuapp.com/icon?url=https://bleacherreport.com/&size=70..120..200"
          style={
            Object {
              "backgroundColor": "#cc0000",
            }
          }
        />
      </div>
      <div
        className="source-title"
      >
        CNN
      </div>
    </a>
    <a
      className="source animated fadeIn"
      href="/vice-news"
      onClick={[Function]}
    >
      <div
        className="source-icon-container"
      >
        <img
          alt="Vice News"
          className="source-icon levitate"
          src="https://besticon-demo.herokuapp.com/icon?url=https://bleacherreport.com/&size=70..120..200"
          style={
            Object {
              "backgroundColor": "#000000",
            }
          }
        />
      </div>
      <div
        className="source-title"
      >
        Vice News
      </div>
    </a>
  </div>
</div>
`;
